src/Microsoft.Extensions.SecretManager.Tools/Internal/DotNetCoreSdkResolver.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using NuGet.Versioning;
+
+namespace Microsoft.Extensions.ProjectModel.Internal
+{
+    internal class DotNetCoreSdkResolver
+    {
+        private readonly string _installationPath;
+
+        public DotNetCoreSdkResolver(string installationDir)
+        {
+            _installationPath = installationDir;
+        }
+
+        private IEnumerable<string> Installed
+            => Directory.EnumerateDirectories(Path.Combine(_installationPath, "sdk"));
+
+        private string ResolveGlobalJsonSdkVersion(string start)
+        {
+            var dir = new DirectoryInfo(start);
+            FileInfo globalJson = null;
+            while (dir != null)
+            {
+                var candidate = Path.Combine(dir.FullName, "global.json");
+                if (File.Exists(candidate))
+                {
+                    globalJson = new FileInfo(candidate);
+                    break;
+                }
+                dir = dir.Parent;
+            }
+            if (globalJson == null)
+            {
+                return null;
+            }
+            var t = new { sdk = new { version = "" } };
```

Comment:
Make a private class, this is super hacky. That will enable you to not declare this here. See next coment